src/main/java/org/spongepowered/api/event/SpongeEventFactory.java

```diff
@@ -189,6 +197,25 @@
     private SpongeEventFactory() {
     }
 
+    private static Class<?> getBaseClass(Class<?> event) {
+        Class<?> superClass = null;
+        for (EventFactoryPlugin plugin: plugins) {
+            superClass = plugin.resolveSuperClassFor(event, superClass);
+        }
+        return superClass;
```

Comment:
Maybe return a default `AbstractEvent` here if `superClass` is still null at this point?